server/server.js

```diff
@@ -1,21 +1,25 @@
 const express = require('express');
 const path = require('path');
+const bodyParser = require('body-parser');
+
 const app = express();
 const port = 3003;
+
 const Reserves = require('../database/Reserve.js');
-const Restaurants = require('../database/Restaurant.js');
 
 app.use(express.static(path.join(__dirname, '../public')));
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: true }));
 
 app.listen(port, () => {
   console.log(`Listening on port: ${port}`);
 });
 
-app.get('/date', (req, res) => {
-  Reserves.find({"date": "2019-04-01T07:00:00.000Z"}).populate("restaurant")
-    .catch((err) => console.error(err))
+app.get('/date/:date', (req, res) => {
+  Reserves.find({ date: req.params.date }).populate('restaurant')
+    .catch(err => console.error(err))
     .then((reservations) => {
       res.send(reservations);
       res.end();
```

Comment:
In express, res.end is not needed. res.send & res.json will both end the resolve.

Reply:
I did not know this! Thank you.